python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -215,18 +212,79 @@ def owner(self) -> Any:
     @property
     def __cuda_array_interface__(self) -> Mapping:
         """Implementation of the CUDA Array Interface."""
+        return self._get_cuda_array_interface(readonly=False)
+
+    def _get_cuda_array_interface(self, readonly=False):
+        """Helper function to create a CUDA Array Interface.
+
+        Parameters
+        ----------
+        readonly : bool, default False
+            If True, returns a CUDA Array Interface with
+            readonly flag set to True.
+            If False, returns a CUDA Array Interface with
+            readonly flag set to False.
+
+        Returns
+        -------
+        dict
+        """
         return {
-            "data": (self.ptr, False),
+            "data": (
+                self.get_ptr(mode="read" if readonly else "write"),
+                readonly,
+            ),
             "shape": (self.size,),
             "strides": None,
             "typestr": "|u1",
             "version": 0,
         }
 
+    @property
+    def _get_readonly_proxy_obj(self):
+        """
+        Returns a proxy object with a read-only CUDA Array Interface.
+        """
+        return cuda_array_interface_wrapper(
+            ptr=self.get_ptr(mode="read"),
+            size=self.size,
+            owner=self,
+            readonly=True,
+            typestr="|u1",
+            version=0,
+        )
+
+    def get_ptr(self, *, mode="write") -> int:
```

Comment:
Maintaining a default was to be on the safe side when someone doesn't specify anything but yeah by making it keyword-only we have eliminated that problem. I don't have a strong preference here, happy to change it. @wence- WDYT? Do you have any preference here?

Reply:
It is currently a no-op, but then COW is enabled. What do we want the default behavior to do? i.e., when someone does `.get_ptr()` call?

Reply:
@vyasr is suggesting this signature:

```python
def get_ptr(self, *, mode) -> int:
    ...
```

This a required keyword-only argument. So it would be invalid to call `buffer.get_ptr()`: the caller must _always_ provide a `mode` argument, but can _only_ do so as a keyword argument.

I think this would be a fine thing to do.

Reply:
Got it, thanks for clarifying. Dropped the default 👍